File: hw/hidalgo/u3/hw35trategyPersistence/java/ToolsManagement/src/main/java/ec/edu/espe/tools/utils/MongoConnection.java
package ec.edu.espe.tools.utils;

import com.mongodb.client.MongoClient;
import com.mongodb.client.MongoClients;
import com.mongodb.client.MongoDatabase;

/**
 *
 * @author Mikael Hidalgo, Paradigm, @ESPE
 */
public class MongoConnection {

    private static MongoConnection instance;
    private MongoClient mongoClient;
    private MongoDatabase database;
    private static final String CONNECTION_STRING = "mongodb+srv://Cesar:Cesar2006@cluster0.tgbv2qc.mongodb.net/";
    private static final String DATABASE_NAME = "ToolsDB";

    private MongoConnection() {
        try {
            mongoClient = MongoClients.create(CONNECTION_STRING);
            database = mongoClient.getDatabase(DATABASE_NAME);
            System.out.println("Sucessful connection");
        } catch (Exception e) {
            System.err.println("Error in connection " + e.getMessage());
        }
    }

    public static MongoConnection getInstance() {
        if (instance == null) {
            instance = new MongoConnection();
        }
        return instance;
    }

    public MongoDatabase getDatabase() {
        return database;
    }
}

File: hw/hidalgo/u3/hw35trategyPersistence/javaScript/ToolsManagement/src/main/javascript/ec/edu/espe/tools/utils/MongoConnection.js
const { MongoClient } = require("mongodb");

class MongoConnection {
  static instance;

  constructor() {
    this.mongoClient = null;
    this.database = null;
  }

  static getInstance() {
    if (!MongoConnection.instance) MongoConnection.instance = new MongoConnection();
    return MongoConnection.instance;
  }

  async connect() {
    if (this.database) return this.database;
    const uri = process.env.MONGO_URI || "mongodb://127mongodb+srv://Cesar:Cesar2006@cluster0.tgbv2qc.mongodb.net/.0.0.1:27017";
    const dbName = process.env.MONGO_DB || "ResourcesDB";
    this.mongoClient = new MongoClient(uri);
    await this.mongoClient.connect();
    this.database = this.mongoClient.db(dbName);
    return this.database;
  }

  async close() {
    if (!this.mongoClient) return;
    await this.mongoClient.close();
    this.mongoClient = null;
    this.database = null;
  }
}

module.exports = MongoConnection;
